src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+    let mainSeparatorClone = separator;
+
+    if (separator === null) {
+      mainSeparatorClone = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === null || this[i] === undefined)
+      && i !== this.length - 1) {
+        resultString += '' + mainSeparatorClone;
+      } else if (this[i] === null || this[i] === undefined) {
+        resultString += '';
+      } else if (i === this.length - 1) {
+        resultString += this[i];
+      } else {
+        resultString += this[i] + mainSeparatorClone;
+      }
```

Comment:
This logic looks too complicated, let's rewrite it using 2 `if` statements. 

To achieve it we need to check: 
1. if our value is not equal to `null` and `undefined` 
2. if it is not a last iteration